thirdparty/github.go

```diff
@@ -81,6 +85,45 @@ const (
 	githubPRUnstable = "unstable"
 )
 
+var cachingTransport http.RoundTripper
+
+type cacheControlTransport struct {
+	base http.RoundTripper
+}
+
+// RoundTrip sets the [max-age] Cache-Control directive before passing the request off
+// to the base transport.
+//
+// [max-age]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#max-age_2
+func (t *cacheControlTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("Cache-Control", "max-age=0")
+	return t.base.RoundTrip(req)
+}
+
+func init() {
```

Comment:
async discussed adding an ADR to explain briefly why we added this, and what different pieces of it do (the span github cache attribute, the different transporters)

Reply:
done